src/Illuminate/Database/Console/PruneCommand.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Illuminate\Database\Console;
+
+use Illuminate\Console\Command;
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Database\Eloquent\MassPrunable;
+use Illuminate\Database\Eloquent\Prunable;
+use Illuminate\Database\Events\ModelsPruned;
+use Illuminate\Support\Str;
+use Symfony\Component\Finder\Finder;
+
+class PruneCommand extends Command
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $signature = 'db:prune {--model=* : Class names of the models to be pruned}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Prune obsolete models';
+
+    /**
+     * Execute the console command.
+     *
+     * @param  \Illuminate\Contracts\Events\Dispatcher  $events
+     * @return void
+     */
+    public function handle(Dispatcher $events)
+    {
+        $events->listen(ModelsPruned::class, function ($event) {
+            $amount = $event->amount;
+            $model = $event->model;
+
+            $this->info("$amount [$model] records have been pruned.");
+        });
+
+        $this->models()->each(function ($model) {
+            $total = $this->isPrunable($model) ? (new $model)->pruneAll() : 0;
+
+            if ($total == 0) {
+                $this->info("No prunable [$model] records found.");
+            }
+        });
+
+        $events->forget(ModelsPruned::class);
+    }
+
+    /**
+     * Determine the models that should be pruned.
+     *
+     * @return array
+     */
+    protected function models()
+    {
+        if (! empty($models = $this->option('model'))) {
+            return collect($models);
+        }
+
+        return collect((new Finder)->in(app_path('Models'))->files())
```

Comment:
What do you think if we elevate the models directory to a command option? Some apps may use a different folder structure - e.g. `app/Domain/Users/User.php` - and listing all models to prune may be tedious.

To speed up the models scanning, we may add the prunable models to a manifest in `bootstrap/cache`.